hs_core/views/__init__.py

```diff
@@ -246,12 +248,39 @@ def delete_file(request, shortkey, f, *args, **kwargs):
 def delete_resource(request, shortkey, *args, **kwargs):
     res, _, _ = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.DELETE_RESOURCE)
 
+    # downgrade collection to private if the res being deleted here is the only member resource in the collection
+    # reverse lookup: find all associated collection resources
+    associated_collection_metadata_obj_list = list(res.collection_set.all())
```

Comment:
This will raise exception when trying to delete any resource other than CollectionResource. This custom logic for CollectionResource needs to be in a signal handler within CollectionResource.